src/pages/basic-concepts/styling.md

```diff
@@ -1,4 +1,510 @@
+
+💁 **Why this works well in Astro**: Astro is inspired most by JavaScript modules: you only need to know about what’s in one file at a time, and you never have to worry about something in a remote file affecting how this code runs. But we’re not alone in this; Vue and Svelte have both capitalized on and popularized the idea that styles and markup are natural fits in the same component file. [You can still have separation of concerns][peace-on-css] even with markup, styling, and logic contained in one file. In fact, that’s what makes component design so powerful! So write CSS without fear that you picked a name that’s used by some other component across your app.
+
+#### Utility CSS
+
+Recently there has been a debate of all-scoped component styles vs utility-only CSS. But we agree with people like Sarah Dayan who ask [why can’t we have both][utility-css]? Truth is that while having scoped component styles are great, there are still hundreds of times when the website’s coming together when two components just don’t line up _quite_ right, and one needs a nudge. Or different text treatment is needed in one component instance.
+
+While the thought of having perfect, pristine components is nice, it’s unrealistic. No design system is absoutely perfect, and every design system has inconsistencies. And it’s in reconciling these inconsistencies where components can become a mess without utility CSS. Utility CSS is great for adding minor tweaks necessary to get the website out the door. But they also are incomplete on their own—if you’ve ever tried to manage responsive styles or accessible focus states with utility CSS it can quickly become a mess! **Utility CSS works best in partnership with component (scoped) CSS**. And in order to be as easy as possible to use, Utility CSS should be global (arguably should be your only global CSS, besides maybe reset.css) so you don’t have to deal with imports all willy-nilly.
+
+Some great problems best handled with Utility CSS are:
+
+- [margin](https://github.com/drwpow/sass-utils#-margin--padding)
+- [padding](https://github.com/drwpow/sass-utils#-margin--padding)
+- [text/background color](https://github.com/drwpow/sass-utils#-color)
+- [font size and family](https://github.com/drwpow/sass-utils#%F0%9F%85%B0%EF%B8%8F-font--text)
+- [default element styling](https://github.com/kognise/water.css)
+
+In Astro, we recommend the following setup for this:
+
+```html
+<head>
+  <link rel="stylesheet" href="/styles/global.css" />
+</head>
+```
+
+And in your local filesystem, you can even use Sass’ [@use][sass-use] to combine files together effortlessly:
+
+```
```

Comment:
Would it make sense to add an example of utility CSS in use somewhere around here? Like an iteration on the above scoped CSS <button> that needs the addition of the utility CSS, or an example that uses multiple `<Button/>`s in a list and shows why `margin` for example is good as a utility class?

Reply:
That’s a good idea. While in general I think the docs could explain utility CSS better, I also think there’s so much to talk about there it deserves a larger section. E.g. would `<button class="mt1">` mean anything to people? Would they expect Bootstrap utility CSS? Bulma? Tailwind? (I even encourage writing your own utility classes at one point).

But if we had a larger explainer, then I’d love to reference back to that in more places like this

Reply:
Agreed, I think as we get more of the documentation filled out with existing information, we all can formally meet about how things can be improved and broken up into more digestible pieces of information. A phase 2 sort of deal.